docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+    "of $|x_{j}\\rangle$.\n",
+    "\n",
+    "To facilitate this representation, we need to map the samples from the multivariate\n",
+    "normal distribution to discrete values. The number of values that can be represented\n",
+    "depends on the number of qubits used for the mapping.\n",
+    "Hence, the data resolution is defined by the number of qubits.\n",
+    "If we use $3$ qubits to represent one feature, we have $2^3 = 8$ discrete values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Load the training data\n",
+    "training_data = algorithm_globals.random.multivariate_normal(\n",
+    "    mean=[0.0, 0.0], cov=[[1, 0], [0, 1]], size=1000, check_valid=\"warn\", tol=1e-8, method=\"svd\"\n",
+    ")\n",
+    "# Define minimal and maximal values for the training data\n",
+    "bounds_min = np.percentile(training_data, 5, axis=0)\n",
+    "bounds_max = np.percentile(training_data, 95, axis=0)\n",
+    "bounds = []\n",
+    "for i, _ in enumerate(bounds_min):\n",
+    "    bounds.append([bounds_min[i], bounds_max[i]])\n",
+    "\n",
+    "# Determine data resolution for each dimension of the training data in terms\n",
+    "# of the number of qubits used to represent each data dimension.\n",
+    "data_dim = [3, 3]\n",
+    "\n",
+    "# Pre-processing, i.e., discretization of the data (gridding)\n",
+    "(training_data, _, grid_elements, prob_data) = discretize_and_truncate(\n",
+    "    training_data,\n",
+    "    np.asarray(bounds),\n",
+    "    data_dim,\n",
+    "    return_data_grid_elements=True,\n",
+    "    return_prob=True,\n",
+    "    prob_non_zero=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "We convert data arrays into tensors and create a data loader from our training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "training_data = torch.tensor(training_data, dtype=torch.float)\n",
+    "grid_elements = torch.tensor(grid_elements, dtype=torch.float)\n",
+    "\n",
+    "# Define the training batch size\n",
+    "batch_size = 300\n",
+    "dataloader = DataLoader(training_data, batch_size=batch_size, shuffle=True, drop_last=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Specify Backend\n",
+    "Next, we need to choose a backend that is used to run the quantum generator.\n",
+    "The presented method is compatible with all shot-based backends (qasm, fake hardware, real hardware) provided by Qiskit.\n",
+    "\n",
+    "First, we create a quantum instance for the training where the batch size defines the number of shots."
```

Comment:
I guess that this will have to be migrated to the primitives at some point too, but because it depends on the QNNs being migrated, maybe it's ok to leave it as a quantum-instance based tutorial for the time being. @adekusar-drl, what do you think? I would propose merging it first "as-is", and then I open a quick PR to use the new primitives-based QNNs instead of `CircuitQNN`.

Reply:
Now, I believe, it is obvious. We need sampling capabilities, so we'll keep the tutorial as it is now.